src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorThreadContainer.java

```diff
@@ -46,13 +46,18 @@ public static MonitorThreadContainer getInstance() {
     public static synchronized MonitorThreadContainer getInstance(IExecutorServiceInitializer executorServiceInitializer) {
         if (singleton == null) {
             singleton = new MonitorThreadContainer(executorServiceInitializer);
+            CLASS_USAGE.getAndSet(0);
         }
-        classUsage.getAndIncrement();
+        CLASS_USAGE.getAndIncrement();
         return singleton;
     }
 
     public static synchronized void releaseInstance() {
-        if (classUsage.decrementAndGet() <= 0) {
+        if (singleton == null) {
+            return;
```

Comment:
Is that the wanted logic? To just return when there is nothing to release?

Is it better than throwing a custom exception to be caught later?